File: tests/interfile/java/test_java_constructor_split/TaintedUser.java
class TaintedUser {
    private String key;

    public TaintedUser(String seller) {
        this.key = source();
    }

    public void props() {
        // ruleid: java_constructor_sqli
        String query = "SELECT * FROM table WHERE name = " + this.key;
        return;
    }
}

File: tests/interfile/typescript/test_typescript_constructor_split/TaintedUser.ts
class TaintedUser {
    private key: string;

    constructor(seller: string) {
        this.key = source();
    }

    props(): void {
        // ruleid: typescript_constructor_sqli
        const query: string = `SELECT * FROM table WHERE name = ${this.key}`;
        return;
    }
}
